tiledb/sm/filesystem/s3.cc

```diff
@@ -586,6 +586,27 @@ Status S3::move_dir(const URI& old_uri, const URI& new_uri) {
   return Status::Ok();
 }
 
+Status S3::copy_file(const URI& old_uri, const URI& new_uri) {
+  RETURN_NOT_OK(init_client());
+
+  RETURN_NOT_OK(copy_object(old_uri, new_uri));
+  return Status::Ok();
+}
+
+Status S3::copy_dir(const URI& old_uri, const URI& new_uri) {
+  RETURN_NOT_OK(init_client());
+
+  std::vector<std::string> paths;
+  RETURN_NOT_OK(ls(old_uri, &paths, ""));
```

Comment:
Please use the default delimiter. This returns all objects in the bucket, which may be a lot. You will need to do a breadth-first traversal like the Posix::copy_dir.